x-pack/plugins/security_solution/server/endpoint/errors.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable max-classes-per-file */
+
+export class EndpointError extends Error {
+  constructor(message: string, public readonly meta?: unknown) {
+    super(message);
+    // For debugging - capture name of subclasses
+    this.name = this.constructor.name;
+  }
+}
+
+export class NotFoundError extends EndpointError {}
```

Comment:
A base class for all "Not Found" errors. Example: see the `EndpointHostNotFoundError` below. Having a base class like this can help in creating generic API route handlers that can evaluate the type of error and respond with the appropriate code (ex. `error instanceof NotFoundError` will work for `TrustedAppNotFoundError` errors as well as `EndpointHostNotFoundError` errors as well as `FleetAgentNotFoundError`'s.